web/src/compose_closed_ui.ts

```diff
@@ -83,9 +90,36 @@ function update_reply_button_state(disable = false): void {
 }
 
 function update_buttons(disable_reply?: boolean): void {
+    if (disable_reply && maybe_get_selected_message_stream_id()) {
+        $("#new_conversation_button").prop("disabled", true);
+    } else {
+        $("#new_conversation_button").prop("disabled", false);
+    }
     update_reply_button_state(disable_reply);
 }
 
+function maybe_get_selected_message_stream_id(): number | undefined {
+    if (message_lists.current?.visibly_empty()) {
+        return undefined;
+    }
+    const selected_message = message_lists.current?.selected_message();
+    if (!selected_message?.is_stream) {
+        return undefined;
+    }
+    return selected_message.stream_id;
+}
+
+function should_disable_compose_reply_button_for_stream(): boolean {
+    const stream_id = maybe_get_selected_message_stream_id();
+    if (stream_id) {
+        const stream = stream_data.get_sub_by_id(stream_id);
```

Comment:
I guess you can just do `narrow_state.stream_sub()` so as to get the current stream subscription.

Reply:
This will not work for some narrow views like `channel: public`, mentions and reactions. They don't have a specific stream but the label of "compose reply button" is set according to the selected message in these views. So we need to check the 
stream the selected message belongs to.

Reply:
oh, ok that makes sense